wsmaster/che-core-api-workspace-shared/src/main/java/org/eclipse/che/api/workspace/shared/dto/event/WorkspaceStatusEvent.java

```diff
@@ -64,4 +64,9 @@
   Map<String, String> getOptions();
 
   WorkspaceStatusEvent withOptions(Map<String, String> options);
+
+  /** @return whether workspace startup was interrupted by user */
+  boolean isStartupInterrupted();
```

Comment:
What about calling this `isInitiatedByUser()`? Because that's what I think we really want to distinguish, isn't it? User-initiated state changes from "environmental" changes in the state.

This way, we could for example also distinguish a user initiated workspace stop from a forced stop due to inactivity (or, God forbid, a user-initiated workspace start from some kind of automated workspace start) without changing the DTO model.

Reply:
interesting idea. I'll take a look.